flang/lib/Optimizer/Transforms/AffineDemotion.cpp

```diff
@@ -86,6 +86,24 @@ class ConvertConversion : public mlir::OpRewritePattern<fir::ConvertOp> {
   }
 };
 
+mlir::Type convertMemRef(mlir::MemRefType type) {
+  return fir::SequenceType::get(
+      SmallVector<int64_t, 8>(type.getShape().begin(), type.getShape().end()),
+      type.getElementType());
```

Comment:
Do MemRefType and SequenceType have the same column/row major aspects ?
Doesn't the affine promotion -> affine demotion lose the SequenceType shape given only `mlir::MemRefType::get({-1}, arrayElementType)` seems to be created in affine promotion [here ](https://github.com/flang-compiler/f18-llvm-project/blob/6bfbede4b5fc4f8f20391b67b2ef4432f05df550/flang/lib/Optimizer/Transforms/AffinePromotion.cpp#L335) ?

Reply:
these changes are for temporary arrays in current affine demotion we don't loose the shape as we just remove the cast. As for difference between memref and sequence type having different column/row major it comes up in lowering and from my understanding should not cause any behavior changes here (that is one reason we convert back to fir).

Reply:
Thanks for this explanation !